#BN-36 feat: Update Profile

Honore123 opened the pull request:
# Ticket Trello Link
[https://trello.com/c/oXlY8e1I/36-update-profile](url)
# PR Description
This PR allows users to update their profiles and make them accessible publicly
# Description of tasks that were expected to be completed
List all the tasks and sub-tasks that were assigned to you Make sure you check every completed task. list them in this style
- [x] The vendor or buyer should be able to update information about their own profile.
- [x] The vendor or buyer's profile should be accessible publicly.
- [x] Appropriate error messages are displayed on unsuccessful attempts.
# Description of tasks that were expected to be completed and that are not yet completed
- [x] None
# How this may be tested?
Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.
1. run `make build` to install all required dependencies
2. visit the `/api-docs` endpoint to access Swagger UI
3. Test profile endpoint `GET` and `PUT`
# Screenshots (If appropriate/applicable)
Provide screenshots related to your PR in case it's appropriate and applicable
# Please check this Checklist before you submit your PR:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My code generates no warnings
- [x] There are no vulnerabilities

Honore123 committed:
feat(profile) create profile feature
- Profile accessible publicly
- Update profile
- Request validation
[Finishes #36]

EvelyneUmubyeyi reviewed (changes requested):
Hello Honore, your endpoints for getting and updating profile work properly. I noticed some few things that you can work on to improve your code:
- Add your endpoints documentation to a separate file in the docs folder
- Moving 'isUserExist' function from the profileController file to another file in the middleware folder
- I think it would be best to return data as an object in the response instead of a list as you did i.e. data: [object]
- There are also no response sent when the user sends an empty request body or non-existant columns on updating profile endpoint

chrisostome01 commented on src/index.js:
```diff
@@ -35,7 +36,7 @@ const specs = swaggerJSDoc(options);
 app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(specs));
 
 app.get('/', (req, res) => res.status(200).json({ status: 200, message: req.t('welcome_message') }));
```
Please checkout the reason why the translation is not working here

chrisostome01 commented on src/database/seeders/20230326221727-demo-cogito.js:
The seeds are not really working, 
[Preview](https://cogito-api-pr-26.onrender.com/api-docs/#/Profile/put_profile__id_)

chrisostome01 replied:
Let also ignore this file in the codeclimate please

chrisostome01 commented on src/database/migrations/20230326212313-create-user.js:
It would be better if this file is not tracked down by codeclimate

chrisostome01 commented on src/controllers/profileController.js:
```diff
@@ -0,0 +1,38 @@
+      .status(200)
+      .json({ status: 200, message: 'Profile updated successfully.', data: updatedProfile[1] });
+  } catch (error) {
+    console.error(error);
```
remove this console

chrisostome01 commented on src/controllers/profileController.js:
```diff
@@ -0,0 +1,38 @@
+  next();
+};
+
+const getProfile = async (req, res) => {
+  const { id } = req.params;
+  const profile = await user.findByPk(id);
+  return res
+    .status(200)
+    .json({ status: 200, message: 'Profile fetched successfully', data: [profile] });
+};
```
Please, implement error handling here too

chrisostome01 commented on src/controllers/profileController.js:
```diff
@@ -0,0 +1,38 @@
+import { user } from '../database/models';
+
+const isUserExist = async (req, res, next) => {
+  const { id } = req.params;
+  const profile = await user.findByPk(id);
+  if (!profile) {
+    res
+      .status(404)
+      .json({ status: 404, message: `user with the id ${id} does not exist`, data: [] });
+  }
+  next();
```
Please, implement error handling here too and move it inside middleware also

Honore123 committed:
fix(profile) fixing profile
- Moving documentation inside docs folder
- Adding codeclimate.yml file
- Adding user.exits middleware
- Creating internationalized response messages
-Adding migration and seeder inside codeclimate.yml
[Finishes #36]

Honore123 committed:
fix(conflicts): conflict with develop
- adding profile swagger file inside apiDocs
- change swagger to yml
- change dev script from node to nodemon
- delete profile file inside docs
[Finishes #36]

chrisostome01 reviewed (approved):
# LGTM

EvelyneUmubyeyi reviewed (approved):
Code meets the desired requirements.
